packages/react/src/components/ContainedList/ContainedList.tsx

```diff
@@ -76,9 +113,14 @@ function ContainedList({
 
   const filteredChildren = filterChildren(children);
 
-  const isActionSearch = ['Search', 'ExpandableSearch'].includes(
-    action?.type?.displayName
-  );
+  const isActionSearch =
+    React.isValidElement(action) &&
+    (typeof action.type === 'string'
+      ? ['Search', 'ExpandableSearch'].includes(action.type)
+      : typeof action.type === 'function' &&
+        'displayName' in action.type &&
+        (action.type.displayName === 'Search' ||
+          action.type.displayName === 'ExpandableSearch'));
```

Comment:
Is there any way to simplify this? It's really dense and hard to read. Is there another spot in the codebase we validate/check displayName/type?

Reply:
I will to simplify this and make it more readable.